### PR TITLE
Add support for AmberTools sqm/out and BondOrderSystem

### PR DESCRIPTION
**Pull request by Ericwang6**
1. Add support for AmberTools sqm output file (a quantum-chemistry software in AmberTools, mainly for AM1-BCC charge calculation)
2. Add BondOrderSystem class, support to read from .mol/.sdf file or rdkit.Chem.Mol object.

### CODE REVIEW

File: dpdata/__init__.py [hunk 1]
```diff
@@ -9,3 +9,16 @@
...
 except ImportError:
     from .__about__ import __version__
+
+# BondOrder System has dependency on rdkit
+try:
+    import rdkit
+    USE_RDKIT = True
+except ModuleNotFoundError:
+    USE_RDKIT = False
+    print("WARNING: No rdkit is installed, 'BondOrderSystem' cannot be used")
```
Comment (njzjz) [inline dpdata/__init__.py:19]: It's not a good idea to show this message to users who don't use this module.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -13,12 +13,18 @@ jobs:
...
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: conda-incubator/setup-miniconda@v2
       with:
-        python-version: ${{ matrix.python-version }}
+        auto-activate-base: true
+        activate-environment: ""
+    # install rdkit and openbabel
+    - name: Install rdkit
+      run: conda create -c conda-forge -n my-rdkit-env python=${{ matrix.python-version }} rdkit openbabel;
```
Comment (njzjz) [inline .github/workflows/test.yml:24]: I don't see where you use openbabel.

Comment (njzjz) [inline .github/workflows/test.yml:24]: You may consider to install https://pypi.org/project/rdkit-pypi/, it should be faster than conda install. Although it is not an official package, we just use it to test the package.
See rdkit/rdkit#1812.

File: dpdata/__init__.py (hunk 1 above)
Comment (Ericwang6) [reply dpdata/__init__.py:19]: OK. I'll remove this.

File: .github/workflows/test.yml (hunk 2 above)
Comment (Ericwang6) [reply .github/workflows/test.yml:24]: Function `dpdata.rdkit.sanitize.convert_by_obabel()` will call obabel via command line. This is designed for the cases where the sanitization fails (using obabel to try to pre-process the mol).

Comment (Ericwang6) [reply .github/workflows/test.yml:24]: I'll see this. But we still have dependency on obabel, which needs conda to install.

File: dpdata/rdkit/sanitize.py [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+    if not os.path.exists(cache_dir):
+        os.mkdir(cache_dir)
+    if mol.HasProp("_Name"):
+        name = mol.GetProp("_Name")
+    else:
+        name = f"mol{int(time.time())}"
+    mol_file_in = os.path.join(cache_dir, f'{name}.mol')
+    mol_file_out = os.path.join(cache_dir, f'{name}_obabel.mol')
+    Chem.MolToMolFile(mol, mol_file_in, kekulize=False)
+    out = os.popen(f"obabel {mol_file_in} -O {mol_file_out}").read()
```
Comment (njzjz) [inline dpdata/rdkit/sanitize.py:459]: Can you consider to use openbabel's python API directly, which should have a better performance?
```suggestion
    from openbabel import openbabel
    obConversion = openbabel.OBConversion()
    obConversion.SetInAndOutFormats("mol", "mol")
    mol = openbabel.OBMol()
    obConversion.ReadFile(mol, mol_file_in)
    obConversion.WriteFile(mol, mol_file_out)
```

**Review by amcadmus — APPROVED** [review]

**Review by njzjz — APPROVED** [review]

**Pull request merged**